### ISSUE TITLE
Publish Docker image

### ISSUE THREAD

**Issue by calebcartwright** [badges/squint#41]
The clean building of the Docker image takes quite a while (as it has to perform a complete build of the application, preferably in Release mode), so it'd be great for us to publish the image, presumably to Dockerhub, so that consumers can simply pull and run the image.

Here's the workflow we have to publish the image for the main Badge Server that should serve as a useful reference

https://github.com/badges/shields/blob/master/.github/workflows/publish-docker-next.yml

**Comment by chris48s** [comment]
A repo under the shieldsio namespace on dockerhub would make the most sense
Do you have access to create one? I don't have sufficient permissions

![Screenshot at 2021-12-05 20-35-14](https://user-images.githubusercontent.com/6025893/144763037-a8d04171-bdb3-4660-a1d1-e8a7aefc6bde.png)

**Comment by calebcartwright** [comment]
Agreed, but unfortunately I don't have access either. Probably something @paulmelnikow will have to create, though also likely makes sense for more than one person to have access (assuming Dockerhub supports it)

**Comment by ghostsquad** [comment]
yes please, any update on this? I'd love to be able to run this, as was reading https://github.com/badges/shields/blob/master/doc/self-hosting.md which lead me here, but there's no easy way to run this raster server.

**Comment by calebcartwright** [comment]
Thanks for sharing your perspective @ghostsquad! Few things I'd note: 

> any update on this?

The absence of shared updates on any badges/* repositories we manage should be explicitly interpreted as meaning there are no updates. While the occasional reminder bump on old issues can serve some utility, there's really no need to ask about updates especially on relatively new threads like this one (we do a pretty good job of sharing updates as and when they become available). The blocker posted above over the last few weeks remains in place.

> there's no easy way to run this raster server.

I'd actually push back on this a little. Docker is the great unifier and simplifier, and the way you'd run this server with docker is exactly the same regardless of where the image comes from. We obviously want to get prebuilt images available on a major registry where it can simply be pulled, but the image can be built trivially by cloning the repo and running canonical docker commands, e.g.

```bash
git clone
cd squint
docker build . ....
docker run ...
```

**Comment by ghostsquad** [comment]
Indeed, it looks like building from source sounds to be pretty trivial. I'm looking forward to getting this up and running, and when I do, I'd be happy to contribute back. Thanks for the reply.

**Comment by calebcartwright** [comment]
> Indeed, it looks like building from source sounds to be pretty trivial. I'm looking forward to getting this up and running, and when I do, I'd be happy to contribute back. Thanks for the reply.

That'd be awesome :rocket: 

Let us know (here and/or on Discord) if you have any issues!

**Comment by calebcartwright** [comment]
Actually, I forgot we hadn't merged the contributing docs (https://github.com/badges/squint/blob/fcb60e4ab812da165ce25401e4eedb2a18677f3e/CONTRIBUTING.md) not exactly thorough yet but may be of some use given the docker section at the bottom

**Comment by XeroxDev** [comment]
Any updates here?

**Comment by chris48s** [comment]
There is now an image you can pull at https://github.com/badges/squint/pkgs/container/squint

At the moment there is one tag `next` which is whatever we've pushed to `main`. I have not looked at tagging releases for numbered versions. Maybe we will at some point, but I have no immediate plans to do that.
